administrator/components/com_workflow/resources/scripts/components/nodes/StageNode.vue

```diff
@@ -0,0 +1,152 @@
+<template>
+  <div
+    class="stage-node card p-3 border shadow-sm position-relative"
+    :class="{ 'shadow': isSelected, 'hover-shadow': !isSelected }"
+    :style="stageStyle"
+    tabindex="0"
+    role="button"
+    :aria-label="`Stage: ${stage?.title}. ${stage?.published ? 'Enabled' : 'Disabled'}`"
+    @keydown="onNodeKeydown"
+    @focus="onNodeFocus"
+    @blur="onNodeBlur"
+    @click="onSelected"
+  >
+    <!-- VueFlow Handles -->
+    <Handle type="target" :position="Position.Top" class="edge-handler bg-primary position-absolute top-0 start-50 translate-middle-x rounded-circle" />
+    <Handle type="source" :position="Position.Bottom" class="edge-handler bg-primary position-absolute bottom-0 start-50 translate-middle-x rounded-circle" />
+    <Handle type="target" :position="Position.Left" class="edge-handler bg-primary position-absolute top-50 start-0 translate-middle-y rounded-circle" />
+    <Handle type="source" :position="Position.Right" class="edge-handler bg-primary position-absolute top-50 end-0 translate-middle-y rounded-circle" />
+
+    <!-- Stage Header -->
+    <div class="card-header d-flex justify-content-between align-items-start p-1">
+      <div class="flex-fill w-75">
+        <h3 class="card-title mb-1 fw-semibold" :title=stage?.title>{{ stage.title }}</h3>
+        <p class="card-text text-muted mb-0 text-truncate" :title=stage?.description>{{ stage.description }}</p>
```

Comment:
Is there any validation/filtering needed? As the description could contain HTML code and could have different length.